### ISSUE TITLE
Different behavior of `map(array)` vs `map(matrix)`

### ISSUE THREAD

**Issue by dvd101x** [josdejong/mathjs#3241]
Hi,

On the current version 13.0.3

By running some tests and changing from Array to Matrix, I'm getting an unexpected error.
```js
math.map([1, 8, 27], math.format) //['1', '8', '27']

math.map(math.matrix([1, 8, 27]), math.format)  // TypeError: Unexpected type of argument in function format (expected: number or BigNumber or function or Object or bigint or Fraction or string or boolean, actual: Array, index: 1)
```

The source of the issue might be that the array implementation is using `applyCallback` but the matrix implementation is using `maxArgumentCount`.

It can be fixed by changing
```js
  'Matrix, function': function (x, callback) {
    return x.map(callback)
  }
```
to
```js
'Matrix, function': (M, callback) => M.create(_map(M.toArray(), callback))
```
But then `Matrix.map` still has the issue in `math.matrix([1, 8, 27]).map(math.format)`

Some of the logic of mapping an array and mapping a matrix is repeated but having a few differences. Maybe there is an opportunity to use a single implementation either in Matrix, in Array or making both of them match.

**Comment by josdejong** [comment]
Published now in `v13.1.0`, thanks again!

**Comment by dvd101x** [comment]
Thank you Jos!
👍
